File: src/flux/model_benchmarks_gpu.jl
include("vggmodels.jl")
include("resnetmodels.jl")
include("benchmarking_gpu.jl")

using Flux, CuArrays

batch_size = 8

run_benchmarks([vgg16, vgg16_bn, vgg19, vgg19_bn, ResNet18, ResNet34, ResNet50], batch_size)

batch_size = 4

run_benchmarks([ResNet101, ResNet152], batch_size)

File: src/pytorch/model_benchmarks_gpu.py
from torchvision.models import vgg16, vgg16_bn, vgg19, vgg19_bn, resnet18, resnet34, resnet50, resnet101, resnet152
from benchmarking_gpu import run_benchmarks

n_iters = 101
batch_size = 8

print("The models are being run on input size of {}".format((batch_size, 3, 224, 224)))
print("The time is the average over {} iterations".format(n_iters-1))
run_benchmarks([vgg16, vgg16_bn, vgg19, vgg19_bn, resnet18, resnet34, resnet50], batch_size, n_iters)

batch_size = 4

print("The models are being run on input size of {}".format((batch_size, 3, 224, 224)))
print("The time is the average over {} iterations".format(n_iters-1))
run_benchmarks([resnet101, resnet152], batch_size, n_iters)
